src/Product.java

```diff
@@ -1,2 +1,105 @@
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.min;
+
 public class Product {
-}
+    private int id;
+
+    private String type;
+    private double price;
+    private boolean discount;
+    private LocalDate addDate;
+
+    public void setPrice(double price) {
+        this.price = price;
+    }
+
+    public static List<Product> getExpensiveBooks
+            (List<Product> products) {
```

Comment:
not good code formatting

Reply:
public static List<Product> filterExpensiveBooks ?
public List<Product> getExpensiveBooks ?